### PR TITLE
Update _minimize.py

### PR DESCRIPTION
**Pull request by aman-thakral**
Adding documentation to specify the form of the optimization problem that is assumed.

### CODE REVIEW

File: scipy/optimize/_minimize.py [hunk 1]
```diff
@@ -38,6 +38,17 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
              callback=None, options=None):
     """
     Minimization of scalar function of one or more variables.
+     
+    In general, the optimization problems are of the form:
+    
+    minimize f(x)
+    subect to:
```
Comment (rgommers) [inline scipy/optimize/_minimize.py]: typo (subect), and needs double colon at the end plus a blank line on the next line. That in order to typeset the indented lines properly in compiled docs.

File: scipy/optimize/_minimize.py [hunk 2]
```diff
@@ -38,6 +38,17 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
...
+    In general, the optimization problems are of the form:
+    
+    minimize f(x)
+    subect to: 
+        g_i(x) >= 0, i = 1,...,m
+        h_i(x)  = 0, i = 1,...,p
+        
+    Where x is a vector of one or more variables.
+    g_i(x) are the inequality constraints.
+    h_i(x) are the equality constrains.
```
Comment (rgommers) [inline scipy/optimize/_minimize.py]: ``x, ``g_i(x)`` and ``h_i(x)`` need to be surrounded by double backticks to get html formatting right. Like:

    ``g_i(x)``

File: scipy/optimize/_minimize.py (hunk 1 above)
Comment (rgommers) [inline scipy/optimize/_minimize.py]: Content: this is only about constraints, I'd mention this explicitly and also add bounds. Something like:

    minimize f(x), optionally within lower/upper bounds and satisfying constraints of the form::

**Comment by rgommers** [comment]
Makes sense to me to add this paragraph.

**Comment by larsmans** [comment]
The commit messages aren't very informative...

**Comment by rgommers** [comment]
guess that's what the Github editor gives you for a default commit message? never tried it so not sure if it can be changed

**Comment by larsmans** [comment]
Is that something we should fix before merging?

**Comment by aman-thakral** [comment]
If it's just the commit message, I can make it more descriptive.

**Comment by rgommers** [comment]
We can. It's a judgement call; in this case squashing the 2 commits and rewriting the commit message would make sense. It's borderline though (if there were 5 commits it wouldn't be), so merging with a descriptive commit message in the merge commit is also possible.

**Comment by rgommers** [comment]
I just saw this explanation: https://guides.github.com/introduction/flow/, and an improvement to it announced today: https://github.com/blog/1945-quick-pull-requests. Looks quite usable.

**Comment by rgommers** [comment]
@aman-thakral can you squash commits in this editor (assuming that's what you're using)?

**Comment by aman-thakral** [comment]
@rgommers I can't seem to squash them in the editor.  I squashed them in git manually.

**Comment by rgommers** [comment]
Good to know. This looks fine now. Minor comment (no need to make a change): we normally use one of the prefixes explained in http://docs.scipy.org/doc/numpy-dev/dev/gitwash/development_workflow.html#writing-the-commit-message. So I'd have started this with ``DOC:``

**Comment by larsmans** [comment]
Merged. Thanks!

**Comment by aman-thakral** [comment]
Great, thanks!

**Pull request merged**